rclcpp/src/rclcpp/subscription_intra_process_base.cpp

```diff
@@ -21,8 +22,9 @@ SubscriptionIntraProcessBase::add_to_wait_set(rcl_wait_set_t * wait_set)
 {
   std::lock_guard<std::recursive_mutex> lock(reentrant_mutex_);
 
-  rcl_ret_t ret = rcl_wait_set_add_guard_condition(wait_set, &gc_, NULL);
-  return RCL_RET_OK == ret;
+  detail::add_guard_condition_to_rcl_wait_set(*wait_set, gc_);
+
+  return true;
```

Comment:
Should we just change this to not return anything?

Reply:
Done in 34f8ed0